### PR TITLE
[DC-2020] secret manager

### PR DESCRIPTION
**Pull request by lrwb-aou**
* initial commit creates the secret manager extension
* secondary commits use the secret manager to obtain and use the mandrill token

### CODE REVIEW

File: data_steward/constants/validation/email_notification.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-MANDRILL_API_KEY = 'MANDRILL_API_KEY'
-UNSET_MANDRILL_API_KEY_MSG = f"Mandrill API key not set in environment variable {MANDRILL_API_KEY}"
+MANDRILL_TOKEN = 'mandrill_token'
```
Comment (dcarbone) [inline data_steward/constants/validation/email_notification.py:1]: Imo lets rename this to something that indicates its the name of a secret.  Maybe something like `MANDRILL_TOKEN_SECRET_ID` or something.  Not

**Review by dcarbone — CHANGES_REQUESTED** [review]
My only request is that the constant be renamed to be more obvious in-name as to what it's value is, but other than that lgtm.

File: data_steward/constants/validation/email_notification.py (hunk 1 above)
Comment (lrwb-aou) [reply data_steward/constants/validation/email_notification.py]: @dcarbone How about now?

**Review by dcarbone — APPROVED** [review]
lgtm!

**Comment by dcarbone** [comment]
The codacy warnings are red herrings and can be ignored.

**Pull request merged**